### PR TITLE
Regex Fixes 1.13

### PR DESCRIPTION
**Pull request by ianmacclancy**
# Description
hand cherrypick changes from [PR: 8143](https://github.com/solo-io/gloo/pull/8143/files)

RouteOptions HostRewriteType_RewritePathRegex does not exist in 1.13 but does in 1.14 onwards so these changes had to be excluded

### CODE REVIEW

**Review by bewebi — DISMISSED** [review]
Guessing buildbot failure is a flake

**Comment by gunnar-solo** [comment]
/kick https://console.cloud.google.com/cloud-build/builds;region=global/fed4a777-cfd1-4ed0-86aa-01bd63119ab1?project=solo-public

**Review by gunnar-solo — DISMISSED** [review]

**Review by sam-heilbron — CHANGES_REQUESTED** [review]

File: changelog/v1.13.17/routeoptions-regex-validation.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo/issues/8091
+    resolvesIssue: false
+    description: >-
```
[the remark this answers is not in the document]
Comment (ianmacclancy) [reply changelog/v1.13.17/routeoptions-regex-validation.yaml:5]: This is called out in the pr description, missed the update to the changelog in the backport

**Review by sam-heilbron — APPROVED** [review]

**Review by bewebi — APPROVED** [review]

**Pull request merged**